replica/table.cc

```diff
@@ -685,7 +691,7 @@ class tablet_storage_group_manager final : public storage_group_manager {
         return storage_group_of(t).first;
     }
     storage_group* storage_group_for_token(dht::token token) const noexcept override {
-        return storage_groups()[storage_group_of(token).first].get();
+        return storage_groups().at(storage_group_of(token).first).get();
```

Comment:
This function is marked noexcept and should return nullptr if the storage_groups doesn't exist.
So better use find() instead of `at`

Reply:
we shouldn't allow nullptr. that should trigger an exception. why is this function marked as noexcept?

Reply:
we should actually abort, because we assert we'll always find storage group allocated.

Reply:
If so, then the function signature can be changed to return a `storage_group&` instead of a pointer.
And this must be documented in `storage_group_manager` api.